packages/remix-dev/compiler.ts

```diff
@@ -417,15 +422,12 @@ async function createServerBuild(
       entryPoints,
       outfile: config.serverBuildPath,
       write: false,
+      conditions: isCloudflareWorkersOrPages ? ["worker"] : undefined,
       platform: config.serverPlatform,
       format: config.serverModuleFormat,
       treeShaking: true,
       minify:
-        options.mode === BuildMode.Production &&
-        !!config.serverBuildTarget &&
-        ["cloudflare-workers", "cloudflare-pages"].includes(
-          config.serverBuildTarget
-        ),
+        options.mode === BuildMode.Production && isCloudflareWorkersOrPages,
       mainFields:
         config.serverModuleFormat === "esm"
           ? ["module", "main"]
```

Comment:
#2076 also changed `mainFields`. Do we want to do that in this PR too?

```suggestion
      mainFields: isCloudflareRuntime
        ? ["browser", "module", "main"]
        : config.serverModuleFormat === "esm"
          ? ["module", "main"]
```

Reply:
@jacob-ebey Do you wanna get this in as well? I think it will be quite helpful but I observed an unexpected behaviour with `esbuild` after expanding the current cf-compiler test suite:

1. If the 3rd party package supports cjs only, i.e.
```json
{
    "name": "browser-pkg",
    "version": "1.0.0",
    "main": "./node-cjs.js",
    "browser": "./browser-cjs.js"
}
```

it bundles `./browser-cjs.js` correctly

2. if the 3rd party package is esm only, i.e.

```json
{
    "name": "browser-pkg",
    "version": "1.0.0",
    "type": "module",
    "main": "./node-esm.js",
    "browser": "./browser-esm.js"
}
```

it bundles `./browser-esm.js`  correctly

3. if the 3rd party package support both cjs and esm, i.e.
```json
{
    "name": "browser-pkg",
    "version": "1.0.0",
    "main": "./node-cjs.js",
    "module": "./node-esm.mjs",
    "browser": {
        "./node-cjs.js": "./browser-cjs.js",
        "./node-esm.mjs": "./browser-esm.mjs"
    }
}
```
it bundles `./node-esm.mjs` instead of `./browser-esm.mjs` this time and changing the platform to `browser` could fix the problem (which was set to `neutral` for cloudflare & deno).

Reply:
@edmundhung, let's change the platform to browser for CF and get this in as well. Looking at the wrangler v2 setup they use the browser: https://github.com/cloudflare/wrangler2/blob/main/packages/wrangler/pages/functions/buildWorker.ts.

Reply:
@jacob-ebey I was trying to reproduce the error when platform is setill set to `netural` but it looks like the tests pass anyway 😓 Do we still wanna update it to `browser`?